Maxis/Infrastructure/Repositories/UserRepository.cs

```diff
@@ -82,23 +84,39 @@ public UserDetailsViewModel ValidateUser(LoginViewModel loginViewModel, bool lda
             {
                 if (ldap)
                 {
-                    
-                    var user = _db.ONNET_USER.FirstOrDefault(u => u.Username == loginViewModel.Username);
-                    if (user != null) return GetRoles(loginViewModel).FirstOrDefault();
-                    var salt = CreateSalt(int.Parse(WebConfigurationManager.AppSettings["salt"]) );
-                    var encyptval = Encrypt(loginViewModel.Password, salt);
-                    var newuser = new ONNET_USER
+                    bool result = ValidateLdap(loginViewModel);
+                    if (result)
                     {
-                        Username = loginViewModel.Username,
-                        Password = salt,
-                        RoleId = (long)Enum.Roles.Normal,
-                        PasswordHash = encyptval
-                    };
-
-                    _db.ONNET_USER.Add(newuser);
-                    _db.SaveChanges();
+                        var user = _db.ONNET_USER.FirstOrDefault(u => u.Username == loginViewModel.Username);
+                        if (user != null) return GetRoles(loginViewModel).FirstOrDefault();
+                        else
+                        {
+                            var salt = CreateSalt(int.Parse(WebConfigurationManager.AppSettings["salt"]));
+                            var encyptval = Encrypt(loginViewModel.Password, salt);
+                            DirectoryEntry ResultPropCall = LoadUserInfo(loginViewModel.Username);
+                            var newuser = new ONNET_USER
+                            {
+                                Username = loginViewModel.Username,
+                                Password = salt,
+                                RoleId = (long)Enum.Roles.Normal,
+                                PasswordHash = encyptval,
+                                Mobile = (ResultPropCall.Properties["mobile"][0].ToString() != null ? ResultPropCall.Properties["mobile"][0].ToString() : ""),
```

Comment:
this will break if there is no value, validate for the key and then for the value